src/Analysis/Engine/Impl/Infrastructure/Extensions/EnumerableExtensions.cs

```diff
@@ -60,6 +60,24 @@ public static bool IsNullOrEmpty<T>(this IEnumerable<T> source)
         public static IEnumerable<T> Keys<T, U>(this IEnumerable<KeyValuePair<T, U>> source) => source.Select(GetKey);
         public static IEnumerable<T> ExcludeDefault<T>(this IEnumerable<T> source) => source.Where(i => !Equals(i, default(T)));
 
+        /// <summary>
+        /// Returns the only element of sequence, or the default value of <typeparamref name="T"/>, if sequence has &lt;&gt; 1 elements.
+        /// </summary>
+        public static T OnlyOneOrDefault<T>(this IEnumerable<T> source) {
```

Comment:
There is `Enumerable.SingleOrDefault<T>` method.

Reply:
It has a different semantics. SingleOrDefault throws if there's more than 1 element. This one returns default.

Reply:
The reasoning here is one would use OnlyOneOrDefault when they want to make a decision based on unambiguous analysis result, if there is no expectation of lack of ambiguity.